mscore/prefsdialog.cpp

```diff
@@ -1141,72 +1303,25 @@ void PreferenceDialog::apply()
-            case 0: ticks = MScore::division;       break;
-            case 1: ticks = MScore::division / 2;   break;
-            case 2: ticks = MScore::division / 4;   break;
-            case 3: ticks = MScore::division / 8;   break;
-            case 4: ticks = MScore::division / 16;  break;
-            case 5: ticks = MScore::division / 32;  break;
-            case 6: ticks = MScore::division / 64;  break;
-            case 7: ticks = MScore::division / 128; break;
-            case 8: ticks = MScore::division / 256; break;
-            default: {
-                  qDebug("Unknown index for shortestNote: %d, defaulting to 16th",
-                         shortestNote->currentIndex());
-                  ticks = MScore::division / 4;
-                  }
+      if(uiStyleThemeChanged) {
+            WorkspacesManager::retranslateAll();
+            preferences.setPreference(PREF_APP_WORKSPACE, WorkspacesManager::currentWorkspace()->name());
+            WorkspacesManager::currentWorkspace()->save();
+            emit mscore->workspacesChanged();
             }
-      preferences.setPreference(PREF_IO_MIDI_SHORTESTNOTE, ticks);
-
-      preferences.setPreference(PREF_IMPORT_OVERTURE_CHARSET, importCharsetListOve->currentText());
-      preferences.setPreference(PREF_IMPORT_GUITARPRO_CHARSET, importCharsetListGP->currentText());
-      preferences.setPreference(PREF_SCORE_NOTE_WARNPITCHRANGE, warnPitchRange->isChecked());
 
-      preferences.setCustomPreference<MuseScoreStyleType>(PREF_UI_APP_GLOBALSTYLE, MuseScoreStyleType(styleName->currentIndex()));
-
-      if (languageChanged) {
-            setMscoreLocale(preferences.getString(PREF_UI_APP_LANGUAGE));
-            mscore->update();
-            }
-
-      if (defaultStyle->text() != preferences.getString(PREF_SCORE_STYLE_DEFAULTSTYLEFILE)) {
-            preferences.setPreference(PREF_SCORE_STYLE_DEFAULTSTYLEFILE, defaultStyle->text());
-            MScore::readDefaultStyle(preferences.getString(PREF_SCORE_STYLE_DEFAULTSTYLEFILE));
-            }
-
-      if (partStyle->text() != preferences.getString(PREF_SCORE_STYLE_PARTSTYLEFILE)) {
-            preferences.setPreference(PREF_SCORE_STYLE_PARTSTYLEFILE, partStyle->text());
-            MScore::defaultStyleForPartsHasChanged();
-            }
-      
-      WorkspacesManager::retranslateAll();
-      preferences.setPreference(PREF_APP_WORKSPACE, WorkspacesManager::currentWorkspace()->name());
-      mscore->changeWorkspace(WorkspacesManager::currentWorkspace());
-      emit mscore->workspacesChanged();
-      
-      emit preferencesChanged();
+      emit preferencesChanged(false, uiStyleThemeChanged);
+      uiStyleThemeChanged = false;
       preferences.save();
       mscore->startAutoSave();
+
+      modifiedWidgets.clear();
+      modifiedUiWidgets.clear();
+      modifiedAudioWidgets.clear();
+
+      buttonBox->button(QDialogButtonBox::Apply)->setText("Apply");
+      cout << "Final: " << timer.elapsed() << endl;
```

Comment:
That is a really useful debugging `cout`, but what about removing it from release build?
```
#ifndef Qt_no_debug // this is not the real name of this macro, I just can't recall rigth now
    cout << "Final: " << timer.elapsed() << endl;
#endif
```

Reply:
Better qDebug()

Reply:
You are right. I'll take care of that.